app/repository/pg/user.go

```diff
@@ -29,23 +29,41 @@ func NewRepo(db *sql.DB) *Repo {
 	}
 }
 
-// Add meth implements usecase.UserRepository interface
+// Add method implements usecase.UserRepository interface
 // without even knowing it that allow us to decouple our layers
 // and will make our app flexible and maintainable.
-func (r Repo) Add(user entities.User) (string, error) {
+func (r Repo) AddUser(user entities.User) (string, error) {
 	var id string
 
 	const sqlStatement = `INSERT INTO "user" (first_name, last_name, email, password)
 						  VALUES ($1, $2, $3, $4) RETURNING id`
 
 	if err := r.DB.QueryRow(sqlStatement, user.FirstName, user.LastName, user.Email, hash.SHA256(user.Password)).Scan(&id); err != nil {
 		var pqErr = new(pq.Error)
-		if errors.As(err, &pqErr) && pqErr.Code.Name() == uniqueViolationErrCode {
-			return "", ErrDuplicateEmail
+		if errors.As(err, &pqErr) && pqErr.Code.Name() == globals.UniqueViolationErrCode {
+			return "", globals.ErrDuplicateEmail
 		}
 
 		return "", fmt.Errorf("error inserting into database: %w", err)
 	}
 
 	return id, nil
 }
+
+// FindUser method implements usecase.UserRepository logic
```

Comment:
Normally when you're in this package you don't know about `usecase` package, you're not even importing it here, so `usecase` should not be mentioned in this package.

Reply:
fixed

Reply:
Same for AddUser

Reply:
fixed